pkg/storage/s3/s3.go

```diff
@@ -351,7 +351,7 @@ func (d *driver) RemoveStorage(cr *opapi.ImageRegistry) error {
 		return err
 	}
 	_, err = svc.DeleteBucket(&s3.DeleteBucketInput{
-		Bucket: aws.String(d.Config.Bucket),
+		Bucket: aws.String(cr.Status.Storage.State.S3.Bucket),
```

Comment:
what if it's not populated for some reason?

Reply:
Then the below code would update the condition on the image registry resource with the code & message from aws

Reply:
actually, when you initiate the driver, d.Config.Bucket is set from cr.Storage.S3.Bucket, so using the Status one is probably wrong here.

Reply:
> Then the below code would update the condition on the image registry resource with the code & message from aws

seems silly to ask aws to tear down an empty bucket name and then report AWS' error to the user.  we can do better.

> actually, when you initiate the driver, d.Config.Bucket is set from cr.Storage.S3.Bucket, so using the Status one is probably wrong here.

Well status should represent "the bucket that we actually know exists because we created it".  Spec is "the bucket we intended to create".  So status seems more correct to me.

Reply:
updated